### DISCUSSION TITLE
Webots simulation

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#926] [Q&A]
Excuse me, Does anybody use the Webots simulation to build loco position system world ? I want to simulate the Crazyflie use UWB to do indoor position navigation. Does it can work ? Thanks.

**Comment by knmcguire** [comment]
Hi! this has not been implemented in Webots yet, but I guess it would be possible! I think it is not 'too' complicated to implement an UWB ranging model in webots and do UWB positioning there. The only think is that we don't have python bindings yet from the crazyflie firmware for the kalman estimator, which would make it even better.

I believe that either SIM_cf or crazyS simulator has a model for the loco positioning system 🤔 It's in gazebo but perhaps you can take some inspiration from that? 

https://www.bitcraze.io/development/external-projects/#crazyflie-simulators

Would be indeed a very nice contribution to webots in the future!
